src/index.ts

```diff
@@ -1,3 +1,24 @@
-import app from './app';
+import cors from '@koa/cors';
+import Koa from 'koa';
+import logger from 'koa-logger';
+import InMemoryArticles from './adaptors/in-memory-articles';
+import App, { AppContext, AppState } from './app';
+import apiDocumentation from './middleware/api-documentation';
+import errorHandler from './middleware/error-handler';
+import routing from './middleware/routing';
+import createRouter from './router';
 
-app.listen(8080);
+const app: App = new Koa<AppState, AppContext>();
+
+app.context.articles = new InMemoryArticles();
+app.context.router = createRouter();
+
+app.use(logger());
+app.use(cors({
+  exposeHeaders: ['Link'],
+}));
+app.use(apiDocumentation());
+app.use(errorHandler());
+app.use(routing(app.context.router));
+
+export default app.listen(8080);
```

Comment:
This change should be extracted.

Reply:
https://github.com/libero/article-store/pull/80 approved and will make this PR leaner.

Reply:
Actually doesn't look like the same thing, but will surely conflict?

Reply:
Just pushed https://github.com/libero/article-store/pull/60/commits/6be8217544419a4127648caf4f6abda4bc96a82a which updates this PR to include #80.